service/src/main/java/com/fecklessweasel/service/FileDownloadServlet.java

```diff
@@ -0,0 +1,80 @@
+package com.fecklessweasel.service;
+
+import com.fecklessweasel.service.datatier.SQLInteractionInterface;
+import com.fecklessweasel.service.datatier.SQLSource;
+import com.fecklessweasel.service.objectmodel.OMUtil;
+import com.fecklessweasel.service.objectmodel.ServiceException;
+import com.fecklessweasel.service.objectmodel.StoredFile;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+/**
+ * Downloads the given file for the user.
+ * @author James Flinn
+ */
+@WebServlet("/servlet/file/download")
+public class FileDownloadServlet extends HttpServlet {
+
+    private static final String FILEPATH_PREFIX = "files";
+
+    /**
+     * When this servlet receives a GET request, downloads the given file.
+     * @param request The web request.
+     * @param response The web response.
+     */
+    @Override
+    protected void doGet(final HttpServletRequest request,
+                         HttpServletResponse response)
+            throws ServletException, IOException {
+
+        final int fid = Integer.parseInt(request.getParameter("fid"));
+
+        String filePath = FILEPATH_PREFIX + "/" + fid;
+        File file = new File(filePath);
+        FileInputStream fileInputStream = new FileInputStream(file);
```

Comment:
Can we make the looking up the inputstream part of the StoredFile lookup method? The point of my refactors was to encapsulate all things files so that all path calculation is one place. That way, when we move the files to an absolute path we have only one place to change it.